google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerOptions.java

```diff
@@ -1081,6 +1085,25 @@ public Builder setCompressorName(@Nullable String compressorName) {
       return this;
     }
 
+    /**
+     * Sets the {@link DirectedReadOptions} that specify which replicas or regions should be used
+     * for non-transactional reads or queries.
+     *
+     * <p>Note that DirectedReadOptions can only be set for ReadOnlyTransaction and
+     * SingleUseTransaction; it cannot be set for ReadWriteTransaction and
+     * PartitionedDmlTransaction.
+     *
+     * <p>DirectedReadOptions set at the request level will take precedence over the options set
+     * using this method. If DirectedReadOptions are set at the request level, they will override
+     * the options set here when the RPC call is made.
+     */
+    public Builder setDirectedReadOptions(DirectedReadOptions directedReadOptions) {
+      Preconditions.checkNotNull(directedReadOptions, "DirectedReadOptions cannot be null");
+      SpannerUtil.verifyDirectedReadOptions(directedReadOptions);
```

Comment:
nit: Could we let the `SpannerUtil#verifyDirectedReadOptions` method return the options that it was given if they are valid? That way the assignment can be on the same line. (I.e. the same as `Preconditions.checkNotNull(..)`)

Reply:
Done.